operator/v1/types_console.go

```diff
@@ -143,8 +143,133 @@ type Capability struct {
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// SourceType defines the source type of the file reference.
+// +kubebuilder:validation:Enum="ConfigMap"
+// +enum
+type SourceType string
+
+const (
+	// SourceTypeConfigMap represents a ConfigMap source.
+	SourceTypeConfigMap SourceType = "ConfigMap"
+)
+
+// ConfigMapFileReference references a specific file within a ConfigMap.
+type ConfigMapFileReference struct {
+	// name is the name of the ConfigMap.
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+
+	// key is the logo key inside the referenced ConfigMap.
+	// Must consist only of alphanumeric characters, dashes (-), underscores (_), and periods (.).
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="self.matches('.*\\\\.[a-zA-Z]{2,5}$')",message="The ConfigMap key must end with a valid file extension (2 to 5 letters)."
+	// +required
+	Key string `json:"key"`
+}
+
+type FileReferenceSource struct {
+	// from specifies the source type of the file reference.
+	// This field is an enum with a valid value "ConfigMap".
+	// +required
+	From SourceType `json:"from"`
+
+	// configMap specifies the ConfigMap source details.
+	// This field is required if From is "ConfigMap".
+	// +optional
+	ConfigMap *ConfigMapFileReference `json:"configMap"`
+}
+
+// Theme defines a theme mode for the console UI.
+type Theme struct {
+	// mode is used to specify what theme mode a logo will apply to in the console UI.
+	// mode is a required field that allows values of Dark and Light.
+	// When set to Dark, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Dark mode.
+	// When set to Light, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Light mode.
+	// +required
+	Mode ThemeMode `json:"mode"`
+
+	// source is used by the console to locate the specified file containing a custom logo.
+	// Source is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension for the dark or light theme so that the console serves the file with the correct MIME type.
```

Comment:
The "for the dark or light theme" is still something that confuses me as a reader and doesn't make me think along the lines of `.png`, `.jpg`, `.svg`, etc. Instead I think "what is a dark or light theme file extension?".
```suggestion
	// The ConfigMap key must include the file extension so that the console serves the file with the correct MIME type.
```